File: leetcode/python/215.kth-largest-element-in-an-array.py
import heapq


class Solution:
    def findKthLargest(self, nums: List[int], k: int) -> int:
        arr = []
        for n in nums:
            heapq.heappush(arr, n)
            if len(arr) > k:
                heapq.heappop(arr)

        return arr[0]

File: leetcode/cpp/215.kth-largest-element-in-an-array.cpp
#include "oj_header.h"

class Solution {
public:
    // O(nlog(k)) 比修改前的版本更快
    int findKthLargest(vector<int>& nums, int k)
    {
        priority_queue<int, vector<int>, greater<int>> pq;

        for (int i : nums) {
            pq.push(i);

            if (pq.size() > k) {
                pq.pop();
            }
        }
        return pq.top();
    }
};
